### PR TITLE
Support for ZeroTier 1.2.4 based on PR https://github.com/alpinelinux/aports/pull/1379

### PR DESCRIPTION
**Pull request by leleobhz**
…/aports/pull/1379

Tested with Raspberry PI with all sugested fixes from refered PR and with ASM disabled due Textrels.

### CODE REVIEW

File: testing/zerotier-one/APKBUILD [hunk 1]
```diff
@@ -1,38 +1,51 @@
...
+## Thanks to Kyle Parisi <kyleparisi@gmail.com> for doing the original package and getting it into testing
+
+## Small changes by Leonardo Amaral <alpinelinux@leonardoamaral.com.br>
+
 pkgname=zerotier-one
-pkgver=1.1.14
+pkgver=1.2.4
 pkgrel=1
-pkgdesc="ZeroTier One allows systems to join and participate in ZeroTier virtual networks."
+pkgdesc="Join ZeroTier virtual neteworks from your Alpine Linux system"
```
Comment (ncopa) [inline testing/zerotier-one/APKBUILD]: "neteworks" looks like typo

Comment (leleobhz) [inline testing/zerotier-one/APKBUILD]: I've fixed it @ncopa - Thanks!

File: testing/zerotier-one/zerotier-one.post-deinstall [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+deluser --remove-home zerotier-one
```
Comment (jirutka) [inline testing/zerotier-one/zerotier-one.post-deinstall]: Don’t do that, NEVER!

File: testing/zerotier-one/zerotier-one.initd [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
-    before firewall netmount
-}
-
 command="/usr/sbin/zerotier-one"
+command_args="$ZEROTIER_OPTS"
 command_background="yes"
-pidfile="/run/$RC_SVCNAME.pid"
+
+start_stop_daemon_args=""
+pidfile="/var/lib/zerotier-one/zerotier-one.pid"
```
Comment (jirutka) [inline testing/zerotier-one/zerotier-one.initd]: This is wrong, pidfiles should be always in `/run`.

File: testing/zerotier-one/zerotier-one.pre-install [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+# Do not fail if user exists, since lbu backup can be restored in diskless with /var/lib/zerotier-one credentials
+
+id zerotier-one >/dev/null 2>&1 || /usr/sbin/adduser -D -H -S -h /var/lib/zerotier-one -G netdev zerotier-one
```
Comment (jirutka) [inline testing/zerotier-one/zerotier-one.pre-install]: This is unnecessary complicated, please follow the same pattern as used in other packages.

```sh
#!/bin/sh

adduser -D -H -S -h /var/lib/zerotier-one -s /sbin/nologin -G netdev -g zerotier-one zerotier-one 2>/dev/null

exit 0
```

File: testing/zerotier-one/zerotier-one.initd [hunk 5]
```diff
@@ -1,11 +1,13 @@
 #!/sbin/openrc-run
 
-depend() {
-    provide net
-    use network
-    before firewall netmount
-}
-
 command="/usr/sbin/zerotier-one"
+command_args="$ZEROTIER_OPTS"
```
Comment (jirutka) [inline testing/zerotier-one/zerotier-one.initd]: This is unnecessary, you can define `command_args` in conf.d/zerotier-one.

File: testing/zerotier-one/zerotier-one.initd [hunk 6]
```diff
@@ -1,11 +1,13 @@
...
 command_background="yes"
-pidfile="/run/$RC_SVCNAME.pid"
+
+start_stop_daemon_args=""
```
Comment (jirutka) [inline testing/zerotier-one/zerotier-one.initd]: Why do you declare empty `start_stop_daemon_args`?

File: testing/zerotier-one/APKBUILD [hunk 7]
```diff
@@ -1,38 +1,51 @@
...
+	zerotier-one.confd
+	${pkgname}-${pkgver}.tar.gz::https://github.com/zerotier/ZeroTierOne/archive/${pkgver}.tar.gz
+	miniupnpc-system.patch
+	doc-build-bash-to-sh.patch
 	"
+builddir="$srcdir/ZeroTierOne-${pkgver}"
+
+prepare() {
+	default_prepare
+}
```
Comment (jirutka) [inline testing/zerotier-one/APKBUILD]: This is redundant. `default_prepare` is implicitly called when you do _not_ define custom `prepare` function.

File: testing/zerotier-one/APKBUILD [hunk 8]
```diff
@@ -1,38 +1,51 @@
-# Contributor: Kyle Parisi <kyleparisi@gmail.com>
-# Maintainer: Kyle Parisi <kyleparisi@gmail.com>
+# Contributor: ZeroTier, Inc. <contact@zerotier.com>
+# Maintainer: Adam Ierymenko <adam.ierymenko@zerotier.com>
+
+## Thanks to Kyle Parisi <kyleparisi@gmail.com> for doing the original package and getting it into testing
+
+## Small changes by Leonardo Amaral <alpinelinux@leonardoamaral.com.br>
```
Comment (jirutka) [inline testing/zerotier-one/APKBUILD]: Please, abuild is not an diary. There may be multiple `Contributor` lines, so just add the second one with your contact.
Also `Maintainer` and `Contributor` should be a real person, not a company.

**Comment by leleobhz** [comment]
Hello!

Time here got a bit depleted last months, but another issue I had was fixed in GIT upstream: https://github.com/zerotier/ZeroTierOne/issues/617

My fork applied this patch, as informed in the bug, so this will not be anymore a issue and dynamic linking with miniupnp and natpmp will be seamlessly possible.

I'll in a few weeks, return the point checked in this PR and fix these issues.

Thanks for you attention!

**Pull request closed without merging**